### PR TITLE
S2712

### PR DESCRIPTION
**Pull request by yeunga**
The changes included here were done before we decided not to make changes to the test scripts. The changes here involve making it easier to switch testing between vault and cavern.

### CODE REVIEW

File: vos/vos/tests/test_vos.py [hunk 1]
```diff
@@ -472,7 +472,8 @@ def test_transfer_error(self, mock_vofile):
 
         vospace_url = 'https://somevospace.server/vospace'
         session = Mock()
-        session.get.side_effect = [Mock(content='COMPLETED')]
+        session.get.side_effect = [Mock(text='COMPLETED'),
+                                   Mock(text='COMPLETED')]
```
Comment (andamian) [inline vos/vos/tests/test_vos.py:476]: Why twice? Is it calling phase twice every time or am I'm missing something?

File: vos/vos/tests/test_vos.py [hunk 2]
```diff
@@ -485,44 +486,47 @@ def test_transfer_error(self, mock_vofile):
         vofile.read.side_effect = [b'QUEUED', b'COMPLETED']
```
Comment (andamian) [inline vos/vos/tests/test_vos.py:486]: Is this still needed?

File: vos/vos/tests/test_vos.py [hunk 3]
```diff
@@ -485,44 +486,47 @@ def test_transfer_error(self, mock_vofile):
...
         self.assertFalse(test_client.get_transfer_error(
             vospace_url + '/results/transferDetails', 'vos://vospace'))
-        session.get.assert_called_once_with(vospace_url + '/phase',
-                                            allow_redirects=False)
+        session.get.assert_called_with(vospace_url + '/phase',
+                                       allow_redirects=False)
```
Comment (andamian) [inline vos/vos/tests/test_vos.py:490]: Again, why twice?

File: vos/vos/tests/test_vos.py [hunk 4]
```diff
@@ -485,44 +486,47 @@ def test_transfer_error(self, mock_vofile):
...
-            [call(vospace_url + '/phase', allow_redirects=False)],
+            [call(vospace_url + '/phase', allow_redirects=False),
+             call(vospace_url + '/phase', allow_redirects=False)],
             session.get.call_args_list)
 
         # job encountered an internal error
         session.reset_mock()
-        vofile.read.side_effect = Mock(side_effect=[b'QUEUED', b'ERROR'])
-        session.get.side_effect = [Mock(content=b'COMPLETED'),
+        session.get.side_effect = [Mock(text='COMPLETED'),
```
Comment (andamian) [inline vos/vos/tests/test_vos.py:507]: ERROR after COMPLETED?

File: vos/vos/tests/test_vos.py (hunk 3 above)
Comment (yeunga) [reply vos/vos/tests/test_vos.py:490]: Again yes.

File: vos/vos/tests/test_vos.py (hunk 4 above)
Comment (yeunga) [reply vos/vos/tests/test_vos.py:507]: Different get calls.

File: vos/vos/tests/test_vos.py (hunk 2 above)
Comment (yeunga) [reply vos/vos/tests/test_vos.py]: Nope, removed.

**Pull request merged**